### PR TITLE
Remote Debug Kogito Operator using Intellij IDEA

### PR DESCRIPTION
**Pull request by vaibhavjainwiz**
Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-cloud-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [ ] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Review by sutaakar — APPROVED** [review]

**Review by ricardozanini — COMMENTED** [review]
Nice doc @vaibhavjainwiz! Many thanks! Alghtough I think it will be better to rephrase [this doc](https://github.com/kiegroup/kogito-cloud-operator/blob/master/docs/GUIDE_TO_KOGITO_DEVS.md) and include this section on it.

**Comment by vaibhavjainwiz** [comment]
> Nice doc @vaibhavjainwiz! Many thanks! Although I think it will be better to rephrase [this doc](https://github.com/kiegroup/kogito-cloud-operator/blob/master/docs/GUIDE_TO_KOGITO_DEVS.md) and include this section on it.

ok.. I do it

**Comment by Kaitou786** [comment]
@vaibhavjainwiz, I think the procedure is same for goland, also do you think we can include some pics.
Plus, alot of developers uses VSCode as well, I think we can include the these steps for VS Code, [see](https://github.com/golang/vscode-go/blob/master/docs/debugging.md) :)

File: README.md [hunk 1]
```diff
@@ -378,6 +379,47 @@ You can use the following command to vet, format, lint, and test your code:
...
+$ export NAMESPACE=default
+$ ./hack/install-manifests.sh
+```
+
+**Start Operator in remote debug mode:**
+
+```bash
+$ cd cmd/manager
+$ export WATCH_NAMESPACE=default
+$ dlv debug --headless --listen=:2345 --api-version=2
```
Comment (spolti) [inline README.md:404]: from where this command is coming from? go-delve/delve?
might be good to specify it.

**Review by spolti — CHANGES_REQUESTED** [review]

**Comment by Kevin-Mok** [comment]
> @vaibhavjainwiz, I think the procedure is same for goland, also do you think we can include some pics?
> Plus, alot of developers uses VSCode as well, I think we can include these steps for VS Code, [see](https://github.com/golang/vscode-go/blob/master/docs/debugging.md?rgh-link-date=2020-10-27T11%3A51%3A23Z) ? :)

@austincunningham (another Red Hatter) has written a couple articles on how to debug an operator in [VSCode](https://dev.to/austincunningham/debug-kubernetes-operator-sdk-locally-using-vscode-130k) and [Goland](https://dev.to/austincunningham/debug-kubernetes-operator-sdk-locally-in-goland-kl6). Maybe we can link to those articles?

File: README.md (hunk 1 above)
Comment (vaibhavjainwiz) [reply README.md:404]: done

**Comment by vaibhavjainwiz** [comment]
> > @vaibhavjainwiz, I think the procedure is same for goland, also do you think we can include some pics?
> > Plus, alot of developers uses VSCode as well, I think we can include these steps for VS Code, [see](https://github.com/golang/vscode-go/blob/master/docs/debugging.md?rgh-link-date=2020-10-27T11%3A51%3A23Z) ? :)
> 
> @austincunningham (another Red Hatter) has written a couple articles on how to debug an operator in [VSCode](https://dev.to/austincunningham/debug-kubernetes-operator-sdk-locally-using-vscode-130k) and [Goland](https://dev.to/austincunningham/debug-kubernetes-operator-sdk-locally-in-goland-kl6). Maybe we can link to those articles?

I had added these link..thanks

**Comment by vaibhavjainwiz** [comment]
> @vaibhavjainwiz, I think the procedure is same for goland, also do you think we can include some pics?
> Plus, alot of developers uses VSCode as well, I think we can include these steps for VS Code, [see](https://github.com/golang/vscode-go/blob/master/docs/debugging.md) ? :)

I had added some screenshot with installation steps. Thanks for your feedback:)

**Comment by vaibhavjainwiz** [comment]
> Nice doc @vaibhavjainwiz! Many thanks! Although I think it will be better to rephrase [this doc](https://github.com/kiegroup/kogito-cloud-operator/blob/master/docs/GUIDE_TO_KOGITO_DEVS.md) and include this section on it.

@ricardozanini **GUIDE_TO_KOGITO_DEVS** is more focused to help kogito service development team so that they don't need to go inside kogito-operator stuff. I think its better to keep it here because we already have article about **Running the Kogito Operator locally** here only

**Review by ricardozanini — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -24,6 +24,7 @@ For information about the Kogito Operator architecture and instructions for usin
       * [Running performance tests](#running-performance-tests)
       * [List of test tags](#list-of-test-tags)
     * [Running the Kogito Operator locally](#running-the-kogito-operator-locally)
+    * [Remote Debug Kogito Operator using Intellij IDEA](#remote-debug-kogito-operator)
```
Comment (Kaitou786) [inline README.md:27]:
```suggestion
    * [Remote Debug Kogito Operator using Intellij IDEA](#remote-debug-kogito-operator-using-intellij-idea)
```
Same as heading for the link to work :)

**Review by Kaitou786 — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (vaibhavjainwiz) [reply README.md]: Thanks for pointed it out. I had fixed it now

**Review by Kaitou786 — APPROVED** [review]

**Review by spolti — APPROVED** [review]

**Pull request merged**